src/overlays/actors/ovl_En_Skj/z_en_skj.c

```diff
@@ -39,201 +180,1498 @@ static ColliderCylinderInitType1 sCylinderInit = {
+}
+
+void EnSkj_SetupSpawnDeathEffect(EnSkj* this) {
+    this->backfilpFlag = 1;
+    EnSkj_SetupAction(this, SKJ_ACTON_SPAWN_DEATH_EFFECT);
+}
+
+void EnSkj_SpawnDeathEffect(EnSkj* this, GlobalContext* globalCtx) {
+    Vec3f effectPos;
+    Vec3f effectVel;
+    Vec3f effectAccel;
+    u32 phi_v0;
+
+    phi_v0 = this->alpha - 4;
+
+    if (phi_v0 > 255) {
+        phi_v0 = 0;
+    }
+    this->alpha = phi_v0;
+    this->actor.shape.shadowAlpha = phi_v0;
+
+    effectPos.x = Rand_CenteredFloat(30.0f) + this->actor.world.pos.x;
+    effectPos.y = Rand_CenteredFloat(30.0f) + this->actor.world.pos.y;
+    effectPos.z = Rand_CenteredFloat(30.0f) + this->actor.world.pos.z;
+
+    effectAccel.z = 0.0f;
+    effectAccel.y = 0.0f;
+    effectAccel.x = 0.0f;
+
+    effectVel.z = 0.0f;
+    effectVel.y = 0.0f;
+    effectVel.x = 0.0f;
+
+    EffectSsDeadDb_Spawn(globalCtx, &effectPos, &effectVel, &effectAccel, 100, 10, 255, 255, 255, 255, 0, 0, 255, 1, 9,
+                         1);
+}
+
+void EnSkj_SetupWaitInRange(EnSkj* this) {
+    this->textId = 0x10BC;
+
+    EnSkj_ChangeAnim(this, SKJ_ANIM_WAIT);
+    EnSkj_SetupAction(this, SKJ_ACTION_SARIA_SONG_WAIT_IN_RANGE);
+}
+
+void EnSkj_WaitInRange(EnSkj* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    u16 reactionId;
+
+    // When link pulls out the Ocarina center him on the stump
+    // Link was probably supposed to be pointed towards skull kid as well
+    if (player->stateFlags2 & 0x1000000) {
+        player->stateFlags2 |= 0x2000000;
+        player->unk_6A8 = &sSmallStumpSkullKid.skullkid->actor;
+        player->actor.world.pos.x = sSmallStumpSkullKid.skullkid->actor.world.pos.x;
+        player->actor.world.pos.y = sSmallStumpSkullKid.skullkid->actor.world.pos.y;
+        player->actor.world.pos.z = sSmallStumpSkullKid.skullkid->actor.world.pos.z;
+        EnSkj_TurnPlayer(sSmallStumpSkullKid.skullkid, player);
+        func_8010BD88(globalCtx, 0x22);
+        EnSkj_SetupWaitForSong(this);
+    } else if (D_80B01EA0 != 0) {
+        player->actor.world.pos.x = sSmallStumpSkullKid.skullkid->actor.world.pos.x;
+        player->actor.world.pos.y = sSmallStumpSkullKid.skullkid->actor.world.pos.y;
+        player->actor.world.pos.z = sSmallStumpSkullKid.skullkid->actor.world.pos.z;
+        if ((Player_GetMask(globalCtx) == PLAYER_MASK_SKULL) && !(gSaveContext.itemGetInf[3] & 0x200)) {
+            func_80078884(NA_SE_SY_TRE_BOX_APPEAR);
+            EnSkj_SetupMaskTrade(this);
+        } else {
+            EnSkj_SetupTalk(this);
+        }
+    } else if (!EnSkj_RangeCheck(player, sSmallStumpSkullKid.skullkid)) {
+        EnSkj_SetupResetFight(this);
+    } else {
+        player->stateFlags2 |= 0x800000;
+        if (gSaveContext.itemGetInf[1] & 0x40) {
+            if (gSaveContext.itemGetInf[3] & 0x200) {
+                reactionId = Text_GetFaceReaction(globalCtx, 0x15);
+                this->textId = reactionId;
+                if ((reactionId & 0xFFFF) == 0) {
```

Comment:
I think this should also match
```suggestion
                this->textId = Text_GetFaceReaction(globalCtx, 0x15);
                if (this->textId == 0) {
```